CIP-XXXX/README.md

```diff
@@ -0,0 +1,84 @@
+---
+CIP: XXXX
+Title: Non-segregated Block Body Serialization
+Category: Ledger
+Authors:
+  - Teodora Danciu <teodora.danciu@iohk.io>
+  - Alexey Kuleshevich <alexey.kuleshevich@iohk.io>
+Implementors: N/A
+Status: Proposed
+Discussions:
+  - https://github.com/IntersectMBO/cardano-ledger/issues/5046
+Created: 2025-07-30
+License: CC-BY-4.0
+---
+
+## Abstract
+
+We propose changing the CBOR encoding of a block body from a segregated layout to a plain sequence of transactions.
+Current layout: all transaction bodies are concatenated and encoded first, followed by their witness sets, then followed by auxiliary-data hashes, and finally followed by validity flags.
+Proposed layout: each transaction is serialized in full before the next transaction is written to the stream.
+
+## Motivation: why is this CIP necessary?
+
+Segregated serialization of [Nested transactions](https://github.com/cardano-foundation/CIPs/pull/862) would be challenging both to specify and implement.
+Separating and concatenating components across nested and non-nested transactions introduces complexity that is error-prone and potentially inefficient, as it may require tracking offsets and performing additional buffering and copying at runtime.
+
+Currently, segregated serialization also complicates the logic used by Consensus to estimate transaction sizes when selecting transactions to fit within a block. Switching to a format where full transactions are encoded sequentially would simplify this process.
+
+Lastly, considering - in hindsight - the original motivation for adopting segregated serialization rather than the more natural and predictable way of serializing a list of transactions - it's unclear whether it provides any real benefit.
+The intent was to enable static validation of a transaction without decoding its witnesses. However, this benefit conflicts with the need for strict field evaluation, which is essential to prevent space leaks caused by retaining the original block bytes for an indeterminate period.
+Moreover, the current segregated layout, if it was used for its intended purpose mentioned above, conflicts with incremental decoding of blocks being received over the wire - for which there is a need.
+Our decoders consume bytes on demand until the entire BlockBody structure is exhausted. As a result, the node decodes the full block - including all witness data - in a single pass. The segregated format therefore provides no practical benefit for partial or streaming validation.
+Moreover, time spent doing deserialization amounts to a small fraction of the whole time spent applying the whole block to the ledger state, even when it is done without validating transactions.
```

Comment:
These lines are nicely broken into concepts, but are currently folded into a really big paragraph... could you add a blank line or two in between, so we get 2 or 3 shorter paragraphs with grouped ideas (for readability's sake)?